openadapt/share.py

```diff
@@ -173,6 +174,37 @@ def receive_recording(wormhole_code: str) -> None:
             logger.info(f"deleted {zip_path=}")
 
 
+def upload_recording_to_s3(user_id: str, recording_id: int) -> None:
+    """Upload a recording to an S3 bucket.
+
+    Args:
+        user_id (str): The ID of the user who owns the recording.
+        recording_id (int): The ID of the recording to upload.
+    """
+
+    def _inner() -> None:
+        try:
+            # Export the recording to a zip file
+            zip_file_path = export_recording_to_folder(recording_id)
+
+            # Upload the zip file to the S3 bucket
+            utils.upload_file_to_s3(
+                zip_file_path,
+                {
+                    "user_id": user_id,
+                },
+            )
+
+            # Delete the zip file after uploading
+            if os.path.exists(zip_file_path):
+                os.remove(zip_file_path)
+                logger.info(f"deleted {zip_file_path=}")
+        except Exception as exc:
+            logger.exception(exc)
+
+    Thread(target=_inner).start()
```

Comment:
Should we add some error handling / retrying / reporting around this?

Reply:
Added retries